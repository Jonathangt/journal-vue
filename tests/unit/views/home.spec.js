
import { shallowMount } from '@vue/test-utils';
import Home  from '@/views/Home.vue';

describe('Pruebas en el home view', () => {
    test('Debe de renderizar el compoiente correctamente', () => {
        const wrapper = shallowMount(Home);
        expect(wrapper.html()).toMatchSnapshot();
        // === 
    })

    test('Al hacer click en un boton debe de redirecionar a no-entry', () => {
        const mockRouterPush = {
            push: jest.fn()
        }

        const wrapper = shallowMount(Home, {
            global:{
                mocks:{
                    $router: mockRouterPush
                }
            }
        })

        wrapper.find('button').trigger('click');

        expect(mockRouterPush.push).toHaveBeenCalledWith({ name: 'no-entry'});
    });
    
})