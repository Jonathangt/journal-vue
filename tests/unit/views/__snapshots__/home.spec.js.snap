// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el home view Debe de renderizar el compoiente correctamente 1`] = `
<div class="home"><img alt="Vue logo" src=""></div>
<h1>Hola mundo</h1>
<button class="btn btn-primary">Primary</button>
<button class="btn btn-secondary">Secondary</button>
<button class="btn btn-success">Success</button>
`;
